rust/connlib/tunnel/src/client.rs

```diff
@@ -284,6 +272,38 @@ where
 
         Ok(())
     }
+
+    fn update_dns_mapping(&mut self) -> bool {
+        let Some(config) = &self.role_state.interface_config else {
+            return false;
```

Comment:
This bails out if the interface is not configured yet?
And when we do configure it, we would come back later to set up the DNS anyway.

Reply:
yeah, this is the same as the old behavior, if we don't have an interface config yet there is still nothing we can do anyways